doc/SONiC_YANG_Model_Guidelines.md

```diff
@@ -6,18 +6,86 @@
 
-These YANG models will be used to verify the configuration for SONiC switches, so a library which supports validation of configuration on SONiC Switch must use these YANG Models. List of such Libraries are: 1.) Configuration Validation Library. (CVL).  YANG models, which are written using these guidelines can also be used as User End YANG Models, i.e North Bound configuration tools or CLI can provide config data in sync with these YANG models.
+[SONiC Management Framework](https://github.com/Azure/SONiC/pull/436) uses SONiC YANG models for configuration management in two ways :
+
+* As *custom Northbound Management YANG* in absence of standard YANG model and for configuration validation (CVL uses it).
+* As *CVL YANG* for configuration validation only, when standard YANG model (Openconfig, IETF etc.) is used as Northbound Management YANG. In such case steps 22 to 24 below can be skipped while developing SONiC YANG model.
 
 
 ## Guidelines
 
-### 1. Each primary section of ABNF.json (i.e a dictionary in ABNF.json) for Example, VLAN, VLAN_MEMBER, INTERFACE  in ABNF.json will be mapped to a container in YANG model.
+
+### 1. All schema definitions related to a feature should be written in a single YANG model file. YANG model file is named as 'sonic-{feature}.yang', e.g. for ACL feature sonic-acl.yang is the file name. It is mandatory to define a top level YANG container named as 'sonic-{feature}' i.e. same as YANG model name. For example, sonic-acl.yang should have 'sonic-acl' as top level container. All other definition should be written inside the top level container.
+
+Example :
+####  YANG
+```
+module sonic-acl {
+	container sonic-acl {
+		.....
+		.....
+	}
+}
+```
+
+### 2. Define namespace as "http://github.com/Azure/{model-name}" and define a shorter(preferably within 3-4 letters) namespace prefix. Use the prefix in XPath for referring to any node present in other model.
+
+Example :
+####  YANG
+```
+module sonic-acl {
+	namespace "http://github.com/Azure/sonic-acl";
+	prefix sms;
+	.....
+	.....
+}
+```
+
+### 3. Use 'revision' to record revision history whenever YANG model is changed. Updating revision with appropriate details helps tracking the changes in the model.
+
+Example :
+```
+module sonic-acl {
+	revision 2019-09-02 {
+		description
+			"Added must expression for ACL_RULE_LIST."; // Updated with new change details
+	}
+	
+	revision 2019-09-01 {
+		description
+			"Initial revision.";
+	}
+
+	container sonic-acl {
+		.....
+		.....
+	}
+}
+```
+
+### 4. Each primary section of ABNF.json (i.e a dictionary in ABNF.json) for Example, VLAN, VLAN_MEMBER, INTERFACE  in ABNF.json will be mapped to a container in YANG model.
```

Comment:
in the translation below, the YANG model is not complete.

Reply:
Since INTERFACE table has special case, we can have rather example of VLAN table here.
@praveen-li I hope you are fine with change proposed ?